Monika After Story/game/script-holidays.rpy

```diff
@@ -18,13 +18,84 @@ define mas_one_hour = datetime.timedelta(seconds=3600)
 define mas_three_hour = datetime.timedelta(seconds=3*3600)
 
 init -1 python:
+    def mas_wasFirstValueIn(_verify, year, *keys):
+        """
+        Checks if the first year that _verify was found for the keys provided in history
+        matches the year provided
+
+        IN:
+            _verify - Value to check for
+            year - year to match
+            *keys - string pieces to make a key for history
+
+        OUT:
+            boolean:
+                - True if the first year matches the year provided
+                - False otherwise
+        """
+        return mas_getFirstYearOfValue(_verify, *keys) == year
+
+    def mas_getFirstYearOfValue(_verify, *keys):
+        """
+        Gets the first year which has the entry of _verify in the keys provided
+
+        IN:
+            _verify - value to check for
+            *keys - string pieces of a key to search for
+
+        OUT:
+            If there's a point where the value we're checking for is found, we return the first year that is met.
+            If not found, we return None
+        """
+        archive_value = mas_HistVerify_k([],_verify, *keys)
+
+        #If we actually have the value we're looking for, we get the first year
+        if archive_value[0]:
+            return archive_value[1][0]
+        return None
+
+    def mas_verifyHistAll_k(_verify, *keys):
+        """
+        Checks if the value of _verify for the keys is in history at any point
+
+        IN:
+            _verify - value to check for
+            *keys - string pieces of a key to search for
+
+        OUT:
+            boolean:
+                - True if _verify is in the key built by the provided pieces at all
+                - False otherwise
+        """
+        return mas_HistVerify_k([],_verify, *keys)[0]
+
+    def mas_verifyHistLastYear_k(_verify, *keys):
+        """
+        Checks history for the value of _verify in the key provided last year
+
+        IN:
+            _verify - value to check for
+            *keys - string pieces of a key to search for
+
+        OUT:
+            boolean:
+                - True if _verify is in the key built by the provided pieces last year
+                - False otherwise
+        """
+        return mas_HistVerify_k([datetime.date.today().year-1], _verify, *keys)[0]
+
+    def mas_lastSeenLastYear(ev_label):
```

Comment:
since `mas_lastSeenInYear` is in `event-handler`, this function should be next to it as its closely related.